internal/analytics/analytics.go

```diff
@@ -0,0 +1,119 @@
+package analytics
+
+// Package analytics abstracts analytics handling for the program.
+// Currently implemented for MixPanel.
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type analyticsConfig struct {
+	token            string
+	mixpanelEndpoint string
+	distinctID       string
+	startupTime      int64
+	client           HTTPClient
+	isAura           bool
+}
+
+type analytics struct {
+	disabled bool
+	cfg      analyticsConfig
+}
+
+// for testing purposes - enables dependency injection of http client
+func NewAnalyticsWithClient(mixPanelToken string, mixpanelEndpoint string, client HTTPClient, isAura bool) Service {
+	distinctID := getDistinctID()
+	cfg := analyticsConfig{
+		token:            mixPanelToken,
+		mixpanelEndpoint: mixpanelEndpoint,
+		distinctID:       distinctID,
+		startupTime:      time.Now().Unix(),
+		client:           client,
+		isAura:           isAura,
```

Comment:
is this currently used? if it's for Aura should we wait until Aura changes to see if this is definitely needed? My reasoning is it's harder to remove things later than not adding it in the first place.

Reply:
Yes it is used, 
this info is to check if the user uses an Aura database or not, from @LackOfMorals requirements,
not sure if @LackOfMorals prefer a different name for the property however